### ISSUE TITLE
ERROR: 'pacman' failed to install missing dependencies.

### ISSUE THREAD

**Issue by technotherion** [2Shirt/WizardKit#17]
I'm guessing that there's a dependency in the AUR that's not met in the "installing missing dependencies" step.

Refer to the following screenshot:

![image](https://user-images.githubusercontent.com/13829537/36348602-7fccb2fa-1430-11e8-9f7a-35b69699f4b2.png)

using either ```./Build\ Linux -u``` or ```./Build\ Linux -b```

**Comment by 2Shirt** [comment]
Might be that the sudo elevation timed-out and the the subsequent sudo call wasn't addressed in time. Perhaps I should adjust the process to expand the dependencies to include package specific dependencies. If that happens, I should also split the AUR process into multiple steps:

* 1. install build-time dependencies
* 2. build AUR packages
* 3. check if still elevated _(and prompt if not)_
* 4. and install AUR packages to live environment.

**NOTE**: This would require another dependency file to cover the build-time requirements.

_If I'm wrong then the new design would at least expose the issue more accurately._

**Comment by technotherion** [comment]
It is difficult to say; especially since not very much information is given. I can wait until a new version is made then try again.

**Comment by 2Shirt** [comment]
You can try doing the following while watching for an unexpected re-elevation:
* Reboot the build environment
* Renaming or deleting the `BUILD_LINUX/live` and `BUILD_LINUX/temp` folders
* Rename or delete the existing ISO files in `OUT_LINUX`
* Run `sudo ./Build\ Linux -f`
* Run `./Build\ Linux -p`
* Run `sudo ./Build\ Linux -o`

**Comment by 2Shirt** [comment]
I misread the error before and my earlier assesment was incorrect. I was thinking the error occured in step 4 but the error is in step 2 so the split-build process I laid out wouldn't address the issue.

After review here's the issue as I see it:
1. Either `sudo` timed out while building the AUR packages; or, `Build Linux` was run unelevated.
2. When the script attempted to build `wd719x-firmware` it attempted to install `lhasa` since it's required at build-time.
3. This called `sudo` and the password was not entered within the timeout and sudo returned exit code 1
4. `makepkg` caught this and printed the error above.

I've added the build-time dependencies so they're installed before attempting to build the packages. This should resolve the issue.

**Comment by technotherion** [comment]
Issue appears to still be  present in Wizardkit-1.4.1-rc2. I will attempt to see if I can build the dependencies before running the script again. 

![image](https://user-images.githubusercontent.com/13829537/36809135-4b485b0e-1c84-11e8-8a8d-e3401d034650.png)

**Comment by 2Shirt** [comment]
1. What was the command you used here, `./Build\ Linux -b` or `./Build\ Linux -o`?
2. The error is with pacman which doesn't build packages, it only installs packages.
3. It still looks like a timeout issue. You can test this by allowing pacman to be run without a password.

This can be done by running `sudo visudo` (or `sudo EDITOR=nano visudo` if you prefer nano) and adding the following line at the bottom of the file:
`dev ALL = (root) NOPASSWD: /usr/bin/pacman`

**Comment by technotherion** [comment]
1.  ```./Build\ Linux -b```
3. I will try this.

I also enabled multilib to see if it's part of the issue, however it did not finish with multilib uncommented. I will recomment multilib in ```/etc/pacman.conf``` and try again with your suggestions.

**Comment by 2Shirt** [comment]
I've added a partial workaround for these su/sudo timeouts in commit 51f9f91. This workaround will not catch timeouts caused by `pacman` like those is these sceenshots. I will recheck the dependencies and retest the build process this week.

**Comment by technotherion** [comment]
Just wanted to give you an update that I've been attempting to run the script but have been waiting on sourceforge to come back online for smartmontools-svn

**Comment by 2Shirt** [comment]
I believe that the `pasystray` package is causing the issue. Since audio is not a priority in the Linux environment I'm going to remove all PulseAudio related packages. This will require reconfiguring the environment for an ALSA-only setup but will reduce the dependencies for the build environment.

**Comment by technotherion** [comment]
svn is finally back online. Will try the newest commit.
